src/org/exist/collections/MutableCollection.java

```diff
@@ -1258,6 +1258,9 @@ private void storeXMLInternal(final Txn transaction, final DBBroker broker, fina
         if(info.isCreating()) {
             info.getTriggers().afterCreateDocument(broker, transaction, document);
         } else {
+            final StreamListener listener = broker.getIndexController().getStreamListener(document, StreamListener.ReindexMode.REPLACE_DOCUMENT);
```

Comment:
`IndexController.getStreamListener()` return same listener until it reset by set it to `null`. `IndexController.getStreamListener(DocumentImpl, ReindexMode)` call `setMode` and it set `listener` to `null` when mode change.